c_api/include/taichi/taichi_cuda.h

```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+#ifndef TI_WITH_CUDA
```

Comment:
In general it's not suggested to change the macro definition in a normal header file, because .cpp files who included  `taichi_cuda.h` will have different `TI_WITH_CUDA` values compared to those who did not. This is likely gonna cause  binary mismatch at link stage. 

Personally I'd suggest ask the users to define `TI_WITH_CUDA` in thier CMakeLists.txt - similar to what we have done in taichi-aot-demo

Reply:
(One more issue)
This also gonna invalidate all the `TI_WITH_CUDA` in `src/taichi_llvm_impl.cpp` since it includes `taichi_cuda.h` - all the `TI_WITH_CUDA` will be override to `ON` no matter what we set `TI_WITH_CUDA` in `TAICHI_CMAKE_ARGS`